bundles/framework/myplacesimport/service/MyPlacesImportService.js

```diff
@@ -33,74 +31,223 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplacesimport.MyPlacesImportSer
      * @method getFileImportUrl
      * @return {String}
      */
-    getFileImportUrl: function () {
-        return this.urls.create;
+    getFileImportUrl: function (sourceSrs) {
+        const params = {
+            srs: this.srs
+        };
+        if (sourceSrs) {
+            params.sourceEpsg = `EPSG:${sourceSrs}`;
+        }
+        return Oskari.urls.getRoute('CreateUserLayer', params);
     },
     /**
      * Returns the url used to update layer.
      *
      * @method getEditLayerUrl
      * @return {String}
      */
-    getEditLayerUrl: function () {
-        return this.urls.edit;
+    getEditLayerUrl: function (id, values) {
+        const params = {
+            id,
+            srs: this.srs,
+            locale: JSON.stringify(values.locale),
+            style: JSON.stringify(values.style)
+        };
+        return Oskari.urls.getRoute('EditUserLayer', params);
+    },
+    submitUserLayer: function (values, successCb, errorCb) {
+        const { sourceSrs, locale, style, file } = values;
+        const formData = new FormData();
+        formData.append('locale', JSON.stringify(locale));
+        formData.append('style', JSON.stringify(style));
+        formData.append('file', file);
+
+        fetch(this.getFileImportUrl(sourceSrs), {
+            method: 'POST',
+            body: formData,
+            headers: {
+                'Accept': 'application/json'
+            }
+        }).then(response => {
+            if (!response.ok && response.status !== 400) {
+                // if bad request try to dig error code from json
+                throw Error(response.statusText);
+            }
+            return response.json();
+        }).then(json => {
+            const { error, info } = json;
+            if (error) {
+                this.log.error(error);
+                this._handleErrorResponse(info, errorCb);
+                return;
+            }
+            this._showSuccess('flyout.success', { count: json.featuresCount });
+            this._handleImportedLayer(json);
+            successCb();
+        }).catch(error => {
+            this.log.error(error);
+            this._showError('flyout.error.generic');
+            errorCb();
+        });
+    },
+
+    updateUserLayer: function (layerId, values, successCb, errorCb) {
+        const id = this.getActualId(layerId);
+        fetch(this.getEditLayerUrl(id, values), {
```

Comment:
I would just leave `id` in url and post the other values as form data instead so the URL doesn't contain the whole payload. Thoughts?

Reply:
changed to post style and locale as form data